Components/CommonCore/Source/gov/sandia/cognition/util/PrimitiveVector.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * File:                PrimitiveVector.java
+ * Authors:             Jeremy D. Wendt
+ * Company:             Sandia National Laboratories
+ * Project:             Cognitive Foundry
+ * 
+ * Copyright Apr 17, 2008, Sandia Corporation.
+ * Under the terms of Contract DE-AC04-94AL85000, there is a non-exclusive
+ * license for use of this work by or on behalf of the U.S. Government. 
+ * Export of this program may require a license from the United States
+ * Government. See CopyrightHistory.txt for complete details.
+ * 
+ */
+
+package gov.sandia.cognition.util;
+
+/**
+ * An abstract base class for memory-dense vectors of primitive types that
+ * permit adding new values, altering elements, etc. Subclass for "int",
+ * "double", and other base types.
+ */
+abstract class PrimitiveVector
```

Comment:
Given we have another meaning for "Vector" in the Foundry, can we rename these classes? They look like primitive array lists. Also, they probably should be in the collection package rather than util.